test/java/controllers/KairosDbProxyControllerTest.java

```diff
@@ -172,26 +176,102 @@ public void testQueryRequiresAggregatorOff() {
                 .setStartTime(Instant.now())
                 .setMetrics(ImmutableList.of(metric1Builder.build(), metric2Builder.build()));
 
-        final Http.RequestBuilder request = Helpers.fakeRequest()
-                .method(Helpers.POST)
+        mockQueryMetrics(controller, builder.build(), any());
+    }
+
+    @Test
+    public void testAddMergeAggregatorOn() {
+        final Metric.Builder metric1Builder = new Metric.Builder()
+                .setName("metric1")
+                .setAggregators(ImmutableList.of(new Aggregator.Builder().setName("count").build()));
+        final Metric.Builder metric2Builder = new Metric.Builder()
+                .setName("metric2")
+                .setAggregators(ImmutableList.of(new Aggregator.Builder().setName("filter").build()));
+        final MetricsQuery.Builder builder = new MetricsQuery.Builder()
+                .setStartTime(Instant.now())
+                .setMetrics(ImmutableList.of(metric1Builder.build(), metric2Builder.build()));
+
+        mockQueryMetrics(_controller, builder.build(), builder.setMetrics(ImmutableList.of(metric1Builder.build(),
+                        metric2Builder.setAggregators(ImmutableList.of(
+                                new Aggregator.Builder().setName("merge").build(),
+                                new Aggregator.Builder().setName("filter").build())).build())).build());
+    }
+
+    @Test
+    public void testAddMergeAggregatorOff() {
```

Comment:
Can you include a test around non-merge specific attributes on the top aggregator with sampling do not get attached to the merge aggregator.